### DISCUSSION TITLE
`useSeoMeta` vs `useServerSeoMeta`

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/nuxt#20758] [Questions] [answered]
I am a bit confused with the difference between `useSeoMeta` and `useServerSeoMeta` because I thought that the `useSeoMeta` will also be run on the client and the server. If we build a site with nuxt3 my understanding is that `useServerSeoMeta` will run once and will not update after that. If we would have a blog which is pre rendered, and we navigate to a different post with nuxt link, do we need to use `useSeoMeta` in this case so that when sharing the post the proper meta data is sent along? And does the `useSeoMeta` composable generate the meta tags for a pre rendered site at build time?

**Comment by TheAlexLichter** [comment, marked as answer]
https://github.com/nuxt/nuxt/pull/20656/ added some new documentation for `useSeoMeta` and `useServerSeoMeta`. Does that help already?

> If we build a site with nuxt3 my understanding is that useServerSeoMeta will run once and will not update after that.

It will run only on the server. If it only runs "once" depends on your rendering mode (e.g. pre-rendering vs. SSR)

> If we would have a blog which is pre rendered, and we navigate to a different post with nuxt link, do we need to use useSeoMeta in this case so that when sharing the post the proper meta data is sent along? 

As crawlers always do a new request and don't do SPA-like navigation, `useServerSeoMeta` should be neough 👍 

> And does the useSeoMeta composable generate the meta tags for a pre rendered site at build time?

Yes, it does! Both composables do kind of the same thing, while `useServerSeoMeta` only runs on the server.

**Reply by dbzx10299** [reply]
Thanks!
